Fix for hibernate 6 issue with stackoverflow, during adding to cart or navigating with having something in the cart

oleksiimiroshnyk opened the pull request:
- Fix for hibernate 6 issue with stackoverflow, during adding to cart or navigating with having something in the cart

Fixes: BroadleafCommerce/QA#5111

This is a known issue we found with hibernate 6...
OrderItem has a 2 lazy references sku & product, if such happens that you fetch orderItem before you've fetched referenced sku or product and then try to access product or sku, or fetch them it will fail with stackoverflow

https://discourse.hibernate.org/t/hibernate-6-onetoone-stackoverflow/8178/2
https://hibernate.atlassian.net/browse/HHH-17140

In hibernate 6.3.1 it is also not fixed

Commit:
BroadleafCommerce/QA#5111
- Fix for hibernate 6 issue with stackoverflow, during adding to cart or navigating with having something in the cart

riteshadhikari17 commented:
I am able to add to cart but I cannot complete checkout. Is this related issue or should I open new ticket?
```
2023-10-26T10:15:03.762-05:00  INFO 41757 --- [nio-8443-exec-3] o.b.core.workflow.SequenceProcessor      : Exception ocurred in blPricingWorkflow, executing rollback handlers
2023-10-26T10:15:03.762-05:00 ERROR 41757 --- [nio-8443-exec-3] o.b.core.workflow.DefaultErrorHandler    : An error occurred during the blPricingWorkflow workflow

java.lang.StackOverflowError: null
	at java.base/java.lang.invoke.DirectMethodHandle.allocateInstance(DirectMethodHandle.java:520)
	at org.hibernate.sql.results.graph.collection.internal.DelayedCollectionAssembler.<init>(DelayedCollectionAssembler.java:27)
	at org.hibernate.sql.results.graph.collection.internal.DelayedCollectionFetch.createAssembler(DelayedCollectionFetch.java:46)
	at org.hibernate.sql.results.graph.entity.AbstractEntityInitializer.<init>(AbstractEntityInitializer.java:172)
	at org.hibernate.sql.results.graph.entity.internal.EntityJoinedFetchInitializer.<init>(EntityJoinedFetchInitializer.java:49)
	at org.hibernate.sql.results.graph.entity.internal.EntityFetchJoinedImpl.buildEntityJoinedFetchInitializer(EntityFetchJoinedImpl.java:129)
	at org.hibernate.sql.results.graph.entity.internal.EntityFetchJoinedImpl.lambda$getEntityInitializer$0(EntityFetchJoinedImpl.java:100)
  ....

2023-10-26T10:15:03.763-05:00 ERROR 41757 --- [nio-8443-exec-3] .BroadleafSimpleMappingExceptionResolver : Error caught and handled.:090f7d45-a7f6-4c64-814a-676832400215

org.broadleafcommerce.core.pricing.service.exception.PricingException: Unable to execute pricing for order -- id: 1
	at org.broadleafcommerce.core.pricing.service.PricingServiceImpl.executePricing(PricingServiceImpl.java:44)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:343)
```

oleksiimiroshnyk commented:
just add more details, I think I was able to complete checkout once....

Commit:
BroadleafCommerce/QA#5111
- Fix for hibernate 6 issue with stackoverflow, during adding to cart & checkout or navigating with having something in the cart
